lib/tasks/cleanup.rake

```diff
@@ -0,0 +1,28 @@
+# typed: false
+# frozen_string_literal: true
+
+namespace :cleanup do
+  desc 'Remove unattached files'
+  task uploads: :environment do
+    # Retrieve all blob keys that were created 7 days ago
+    uploaded_blobs = ActiveStorage::Blob.where('DATE(created_at) = ?', 7.days.ago)
+
+    # Retrieve file attachments for the above blobs
+    attached_files = ActiveStorage::Attachment.where(blob_id: uploaded_blobs)
+
+    # Get the ids and keys for blobs that do not have an Attachment
+    unattached_blobs = uploaded_blobs.map(&:id) - attached_files.map(&:blob_id)
+    unattached_file_keys = ActiveStorage::Blob.find(unattached_blobs).map(&:key)
```

Comment:
a few random thoughts:
1. with this version of the PR, i think it'd be a win just switching from `map` to `pluck` (and dropping the `&` -- e.g. `uploaded_blobs.pluck(:id)`).  i believe that'll get you the same result, ultimately, but it'll do it without instantiating an active record object for each DB row, and i think it'll also select solely the column you're interested in, as opposed to all the table's columns (not a notable difference for most queries, but might make a difference when iterating over a potentially large result set like this).
2. iirc from what i saw of yesterday's version, i _think_ the subquery approach could work as a single query if you traded what was a `map` for a `select` -- my recollection is that for subqueries like that, ActiveRecord is smart enough to use the `Relation` from the `select` call to build a single whole SQL query (instead of running the inner select, retrieving all the results, mapping over the result set to get IDs, and then feeding those IDs to the outer query and running that second query separately).
3. another approach that i think is usable as an alternative to the subquery approach would be an [outer join](https://apidock.com/rails/v6.1.3.1/ActiveRecord/QueryMethods/left_outer_joins) to the table where you want to find no entries, with `nil` as the matching condition  on that table.  something like... `ActiveStorage::Blob.left_outer_joins(:active_storage_attachments).where('DATE(active_storage_blobs.created_at) < ?', 7.days.ago).where(blob_id: nil)` -- or something, i'm writing that freehand in a comment box without testing, so grain of salt 🙂 

here's an example where a `not` subquery is built using a scope instead of a concrete retrieved result set:  https://github.com/sul-dlss/preservation_catalog/blob/2dc297dd416f5264902fabb919805cac666a26db/app/models/zip_endpoint.rb#L26-L38

my recollection is i went that way on that pres cat query because i compared the plan explanation for that with the outer join approach, and the outer join approach was less efficient.

if none of the single shot approaches work, you could always fall back to iterating over `uploaded_blobs` using `uploaded_blobs.find_each`, and querying for each blob within that -- unlikely to make anything fall over, but also seems likely to take quite a while to run (and less fun imho than building one nice clean SQL query, but i'm a weirdo who likes SQL 🙂).